proto/protovalidate-testing/buf/validate/conformance/cases/messages.proto

```diff
@@ -58,3 +58,41 @@ message MessageRequiredOneof {
 }
 
 message MessageWith3dInside {}
+
+message MessageOneofSingleField {
+  option (buf.validate.message).oneof = {
+    fields: ["str_field"]
+  };
+  string str_field = 1;
+  bool bool_field = 2;
+}
+
+message MessageOneofMultipleFields {
+  option (buf.validate.message).oneof = {
+    fields: [
+      "str_field",
+      "bool_field"
+    ]
+  };
+  string str_field = 1;
+  bool bool_field = 2;
+}
+
+message MessageOneofMultipleFieldsRequired {
+  option (buf.validate.message).oneof = {
+    fields: [
+      "str_field",
+      "bool_field"
+    ]
+    required: true
+  };
+  string str_field = 1;
+  bool bool_field = 2;
+}
+
+message MessageOneofUnknownFieldName {
```

Comment:
consider also having oneof with no fields (which should also be an error)

Reply:
In light of this, I removed the `MessageOneofNoFields` test case in https://github.com/bufbuild/protovalidate/pull/377/commits/fd2217b1abc85bbf5beb55bc5ec0596bc0e4981f and we can address as a follow-up.